### ISSUE TITLE
CommonInterruptible.interrupt incorrectly assumes it's being called from an interrupted thread

### ISSUE THREAD

**Issue by nicktindall** [OpenHFT/Chronicle-Core#251]
We replace the "interruptor" field on interruptible channels with an implementation that doesn't close the channel when interrupted https://github.com/OpenHFT/Chronicle-Core/blob/f5f12d4c0f1f0b803c861e8cd9d44bf31aa6aad4/src/main/java/net/openhft/chronicle/core/Jvm.java#L1372

This implementation logs a warning if the calling thread is not interrupted
https://github.com/OpenHFT/Chronicle-Core/blob/f5f12d4c0f1f0b803c861e8cd9d44bf31aa6aad4/src/main/java/net/openhft/chronicle/core/Jvm.java#L1541-L1545

One way `Interrubtible#interrupt` is called is when a thread is interrupting a second thread which is blocked in an IO operation. In this scenario `isInterrupted` of the calling thread can be (and probably usually is) false.

In Java 8: https://github.com/openjdk/jdk/blob/9a9add8825a040565051a09010b29b099c2e7d49/jdk/src/share/classes/java/lang/Thread.java#L910

In Java 11:
https://github.com/openjdk/jdk/blob/da75f3c4ad5bdf25167a3ed80e51f567ab3dbd01/src/java.base/share/classes/java/lang/Thread.java#L984

A stack-trace from such an call is below:
```
interrupt:1545, Jvm$CommonInterruptible (net.openhft.chronicle.core)
lambda$doNotCloseOnInterrupt9$6:1404, Jvm (net.openhft.chronicle.core)
invoke:-1, 1063621080 (net.openhft.chronicle.core.Jvm$$Lambda$120)
interrupt:-1, $Proxy9 (com.sun.proxy.jdk.proxy1)
interrupt:988, Thread (java.lang)
interruptIfStarted:663, ThreadPoolExecutor$Worker (java.util.concurrent)
interruptWorkers:761, ThreadPoolExecutor (java.util.concurrent)
shutdownNow:1407, ThreadPoolExecutor (java.util.concurrent)
shutdownNow:693, Executors$DelegatedExecutorService (java.util.concurrent)
shutdownDaemon:112, Threads (net.openhft.chronicle.threads)
shutdown:130, Threads (net.openhft.chronicle.threads)
shutdownService:571, MediumEventLoop (net.openhft.chronicle.threads)
stopEventLoopThread:176, MediumEventLoop (net.openhft.chronicle.threads)
performStopFromStarted:171, MediumEventLoop (net.openhft.chronicle.threads)
stop:62, AbstractLifecycleEventLoop (net.openhft.chronicle.threads)
lambda$addAllEventLoopStoppers$0:46, EventLoops (net.openhft.chronicle.threads)
call:-1, 948034998 (net.openhft.chronicle.threads.EventLoops$$Lambda$509)
exec:1448, ForkJoinTask$AdaptedCallable (java.util.concurrent)
doExec$$$capture:290, ForkJoinTask (java.util.concurrent)
doExec:-1, ForkJoinTask (java.util.concurrent)
tryRemoveAndExec:1062, ForkJoinPool$WorkQueue (java.util.concurrent)
awaitJoin:1688, ForkJoinPool (java.util.concurrent)
doJoin:397, ForkJoinTask (java.util.concurrent)
quietlyJoin:1078, ForkJoinTask (java.util.concurrent)
invokeAll:2519, ForkJoinPool (java.util.concurrent)
stopAll:27, EventLoops (net.openhft.chronicle.threads)
performStop:330, EventGroup (net.openhft.chronicle.threads)
performStopFromStarted:325, EventGroup (net.openhft.chronicle.threads)
stop:62, AbstractLifecycleEventLoop (net.openhft.chronicle.threads)
lambda$addAllEventLoopStoppers$0:46, EventLoops (net.openhft.chronicle.threads)
call:-1, 948034998 (net.openhft.chronicle.threads.EventLoops$$Lambda$509)
exec:1448, ForkJoinTask$AdaptedCallable (java.util.concurrent)
doExec$$$capture:290, ForkJoinTask (java.util.concurrent)
doExec:-1, ForkJoinTask (java.util.concurrent)
topLevelExec:1020, ForkJoinPool$WorkQueue (java.util.concurrent)
scan:1656, ForkJoinPool (java.util.concurrent)
runWorker:1594, ForkJoinPool (java.util.concurrent)
run:183, ForkJoinWorkerThread (java.util.concurrent)
```

The other scenario is when an interrupted thread attempts to interact with the channel, in this case, `isInterrupted` of the calling thread will be true. A stack-trace from such a call is below:
```
interrupt:1542, Jvm$CommonInterruptible (net.openhft.chronicle.core)
lambda$doNotCloseOnInterrupt8$5:1385, Jvm (net.openhft.chronicle.core)
interrupt:-1, 1706178768 (net.openhft.chronicle.core.Jvm$$Lambda$64)
begin:173, AbstractInterruptibleChannel (java.nio.channels.spi)
size:305, FileChannelImpl (sun.nio.ch)
resizeRafIfTooSmall:215, ChunkedMappedFile (net.openhft.chronicle.bytes.internal)
acquireByteStore:168, ChunkedMappedFile (net.openhft.chronicle.bytes.internal)
acquireByteStore:198, MappedFile (net.openhft.chronicle.bytes)
acquireNextByteStore0:321, ChunkedMappedBytes (net.openhft.chronicle.bytes.internal)
writeCheckOffset:259, ChunkedMappedBytes (net.openhft.chronicle.bytes.internal)
compareAndSwapInt:185, AbstractBytes (net.openhft.chronicle.bytes)
writeFirstHeader:401, AbstractWire (net.openhft.chronicle.wire)
acquire:1018, SingleChronicleQueue$StoreSupplier (net.openhft.chronicle.queue.impl.single)
acquire:53, WireStorePool (net.openhft.chronicle.queue.impl)
setCycle2:236, StoreAppender (net.openhft.chronicle.queue.impl.single)
setWireIfNull:434, StoreAppender (net.openhft.chronicle.queue.impl.single)
writeBytesInternal:568, StoreAppender (net.openhft.chronicle.queue.impl.single)
writeBytesInternal:559, StoreAppender (net.openhft.chronicle.queue.impl.single)
appendInternal:30, BytesAppender (net.openhft.chronicle.queue.impl.single)
onRead:258, SinkReplicationHandler (software.chronicle.enterprise.queue.replication)
onRead:221, UberHandler (net.openhft.chronicle.network.cluster.handlers)
onRead0:186, WireTcpHandler (net.openhft.chronicle.network)
process:124, WireTcpHandler (net.openhft.chronicle.network)
invokeHandler:414, TcpEventHandler (net.openhft.chronicle.network)
readAction:268, TcpEventHandler (net.openhft.chronicle.network)
action0:228, TcpEventHandler (net.openhft.chronicle.network)
action:197, TcpEventHandler (net.openhft.chronicle.network)
runAllMediumHandler:356, MediumEventLoop (net.openhft.chronicle.threads)
runLoop:271, MediumEventLoop (net.openhft.chronicle.threads)
run:231, MediumEventLoop (net.openhft.chronicle.threads)
call:511, Executors$RunnableAdapter (java.util.concurrent)
run$$$capture:266, FutureTask (java.util.concurrent)
run:-1, FutureTask (java.util.concurrent)
 - Async stack trace
<init>:151, FutureTask (java.util.concurrent)
newTaskFor:87, AbstractExecutorService (java.util.concurrent)
submit:111, AbstractExecutorService (java.util.concurrent)
submit:678, Executors$DelegatedExecutorService (java.util.concurrent)
performStart:149, MediumEventLoop (net.openhft.chronicle.threads)
start:41, AbstractLifecycleEventLoop (net.openhft.chronicle.threads)
getReplication:181, EventGroup (net.openhft.chronicle.threads)
addHandler:248, EventGroup (net.openhft.chronicle.threads)
action:193, RemoteConnector$RCEventHandler (net.openhft.chronicle.network)
run:161, BlockingEventLoop$Runner (net.openhft.chronicle.threads)
call:511, Executors$RunnableAdapter (java.util.concurrent)
run$$$capture:266, FutureTask (java.util.concurrent)
run:-1, FutureTask (java.util.concurrent)
 - Async stack trace
<init>:151, FutureTask (java.util.concurrent)
newTaskFor:87, AbstractExecutorService (java.util.concurrent)
submit:111, AbstractExecutorService (java.util.concurrent)
startHandler:92, BlockingEventLoop (net.openhft.chronicle.threads)
forEach:1259, ArrayList (java.util)
performStart:87, BlockingEventLoop (net.openhft.chronicle.threads)
start:41, AbstractLifecycleEventLoop (net.openhft.chronicle.threads)
performStart:292, EventGroup (net.openhft.chronicle.threads)
start:41, AbstractLifecycleEventLoop (net.openhft.chronicle.threads)
start:130, Cluster (net.openhft.chronicle.network.cluster)
startReplication:140, ReplicatedQueue (software.chronicle.enterprise.queue.replication)
startReplicator:86, BlockAppendersUntilReplicatingAndFinishedBackfillTest (software.chronicle.enterprise.queue.replication)
shouldBlockAcquireAppenderUntilReplicatorIsRunningAndBackfillIsFinished:53, BlockAppendersUntilReplicatingAndFinishedBackfillTest (software.chronicle.enterprise.queue.replication)
```

**Comment by nicktindall** [comment]
Running QE's `BlockAppendersUntilReplicatingAndFinishedBackfillTest` will hit this branch if run until failure

**Comment by hft-team-city** [comment]
Released in [Chronicle-Core-2.22ea7](https://github.com/OpenHFT/Chronicle-Core/releases/chronicle-core-2.22ea7), [BOM-2.22ea29](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.22ea29)
